manticore/core/cpu/arm.py

```diff
@@ -1066,14 +1066,21 @@ def _STM(cpu, insn_id, base, regs):
         address = base.read()
         if insn_id == cs.arm.ARM_INS_STMIB:
             address += 4
+        if insn_id == cs.arm.ARM_INS_STMDB:
```

Comment:
```suggestion
        elif insn_id == cs.arm.ARM_INS_STMDB:
```

Reply:
I've refactored the `_LDM` and `_STM` methods. Hopefully they are easier to understand now.